File: src/main/java/com/nik7/upgradecraft/datagenerators/DataGenerators.java
package com.nik7.upgradecraft.datagenerators;

import net.minecraft.data.DataGenerator;
import net.minecraftforge.data.event.GatherDataEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;

import static com.nik7.upgradecraft.UpgradeCraft.MODID;

@Mod.EventBusSubscriber(modid = MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
public class DataGenerators {
    @SubscribeEvent
    public static void gatherDataGenerators(GatherDataEvent event) {
        DataGenerator generator = event.getGenerator();


        generator.addProvider(event.includeServer(), new UpgCLootTableProvider(generator));
        generator.addProvider(event.includeServer(), new UpgCRecipeProvider(generator));
        generator.addProvider(event.includeServer(), new UpgCBlockTagsProvider(generator, event.getExistingFileHelper()));
        // generator.addProvider(new ItemTagsProviderUpgC(generator, blockTagsProvider, existingFileHelper));

        generator.addProvider(event.includeClient(), new UpgCBlockStateProvider(generator, event.getExistingFileHelper()));
        generator.addProvider(event.includeClient(), new UpgCItemModelProvider(generator, event.getExistingFileHelper()));
        //lang
        generator.addProvider(event.includeClient(), new UpgCLanguageProvider.US(generator));

    }
}

File: src/main/kotlin/com/nik7/upgradecraft/datagenerators/DataGenerators.kt
@file:Mod.EventBusSubscriber(modid = UpgradeCraft.ID, bus = Mod.EventBusSubscriber.Bus.MOD)

package com.nik7.upgradecraft.datagenerators

import com.nik7.upgradecraft.UpgradeCraft
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.forge.event.lifecycle.GatherDataEvent


@SubscribeEvent
fun gatherDataGenerators(event: GatherDataEvent) {
    val generator = event.generator
    val helper = event.existingFileHelper

    if (event.includeServer()) {
        generator.addProvider(ModRecipeProvider(generator))
        generator.addProvider(ModLootTableProvider(generator))
    }

    if (event.includeClient()) {
        generator.addProvider(ModBlockStateProvider(generator, helper))
        generator.addProvider(ModItemModelProvider(generator, helper))
        generator.addProvider(ModLanguageProvider.UsLanguageProvider(generator))
    }

}
